src/inmanta/ast/statements/assign.py

```diff
@@ -622,13 +622,13 @@ class StringFormatV2(FormattedString):
     def __init__(self, format_string: str, variables: abc.Sequence[typing.Tuple["Reference", str]]) -> None:
         """
         :param format_string: The string on which to perform substitution
-        :param variables: Sequence of tuples each holding a reference to a variable to substitute in the format_string
-            and the name of this variable. This name is not used in practice since it's already present in the reference's
-            full_name attribute.
+        :param variables: Sequence of tuples each holding a clean reference (i.e. stripped of eventual whitespaces ) to a
+         variable to substitute in the format_string and the raw full name of this variable (i.e. including eventual
+         whitespaces).
         """
         only_refs: abc.Sequence["Reference"] = [k for (k, _) in variables]
         super().__init__(format_string, only_refs)
-        self._variables = only_refs
+        self._variables = {ref: full_name for (ref, full_name) in variables}
```

Comment:
```suggestion
        self._variables: abc.Mapping[Reference, str] = {ref: full_name for (ref, full_name) in variables}
```

Could you also move `_variables` to this class' `slots`? I don't think it belongs in the parent anymore.

Reply:
Moved it down from `FormattedString` into  `StringFormat` and `StringFormatV2`. I'm not super familiar with slots usage, iirc it saves memory by not duplicating the field's name into each object ?
Any motivation for moving it down from the parent's class ?

Reply:
It's not used/declared in the class. Having it in slots is like declaring it's part of the interface of the class, but it really isn't in this case. So my motivation in this case is really more about readability than anything else.

Reply:
Makes sense :+1: